mono/metadata/handle.h

```diff
@@ -431,12 +436,11 @@ This is why we evaluate index and value before any call to MONO_HANDLE_RAW or ot
 		(DEST) =  __result;					\
 	} while (0)
 
-#define MONO_HANDLE_ARRAY_GETREF(DEST, HANDLE, IDX) do {		\
-		mono_handle_array_getref (MONO_HANDLE_CAST(MonoObject, (DEST)), (HANDLE), (IDX)); \
-	} while (0)
+#define MONO_HANDLE_ARRAY_GETREF(DEST, HANDLE, IDX) \
+		(mono_handle_array_getref (MONO_HANDLE_CAST(MonoObject, (DEST)), (HANDLE), (IDX)))
```

Comment:
Why changing that from a statement to an expression?

Reply:
Expressions are always preferred, right? Why not? I've never heard otherwise. There is an occasional complaint that C/C++ aren't more expression-based, like if/else could be expressions. They are in other languages. We are stuck with what we have, but make the best of it?

Reply:
What does it bring exactly in this specific case? "Expressions are always preferred" is, without a proper argument, just an assumption.